lib/browser.js

```diff
@@ -121,15 +121,26 @@ class Browser {
 
   reconnect (newSocket) {
     if (this.state === EXECUTING_DISCONNECTED) {
+      // In case the socket just lost connection and the browser continued the execution,
+      // we just update the state back to "EXECUTING" so that it matches the current state.
       this.log.debug(`Reconnected on ${newSocket.id}.`)
       this.setState(EXECUTING)
     } else if ([CONNECTED, CONFIGURING, EXECUTING].includes(this.state)) {
+      // In case for some reason the socket has changed, we just bind the events of the
+      // new socket (see below)
```

Comment:
Rather than the comment, how about just changing the log message:
`Rebinding to new socket ${newSocket.id} (already have ${this.getActiveSocketsIds()})`

Reply:
Same as below. This works for me.